src/Core/Rendering/RenderController.cpp

```diff
@@ -766,6 +766,13 @@ namespace MxEngine
 
         MAKE_SCOPE_PROFILER("RenderController::ApplyLensFlare()");
          
+        float averageLum = 0.f;
+        if (camera.ToneMapping != nullptr)
+        {
+            Image averageTexData = camera.AverageWhiteTexture->GetRawTextureData();
+            averageLum = *reinterpret_cast<float*>(averageTexData.GetRawData());
```

Comment:
thats really bad as introduces implicit gpu -> cpu sync. Read this texture in shader